pkg/kubectl/cmd/util/diff/diffoptions.go

```diff
@@ -121,6 +123,18 @@ func (o *DiffOptions) Complete(f cmdutil.Factory, cmd *cobra.Command) error {
 	return err
 }
 
+func isAnnotationsEmpty(obj runtime.Object) bool {
```

Comment:
You don't need to do it by youself.
We already have some handy tools.
See: https://gist.github.com/mengqiy/61189aeecf13660430471aac0c45a978